app/controllers/welcome_controller.rb

```diff
@@ -0,0 +1,29 @@
+class WelcomeController < ApplicationController
+  before_action :current_user
+  include VimeoWrapper
+
+  def index
+    if @current_user.nil?
+      redirect_to login_path
+    else
+      @feed = make_feed(@current_user)
+    end
+  end
+
+  def make_feed(current_user)
```

Comment:
This is a great idea to move the functionality of making the feed out of the index controller! Bonus points for that! 

My only suggestion: it looks like what you're doing in this method is restructuring data that you get from your database. This is a perfect example of where you can use a model scope/method to keep all your data structuring in your model and out of your controller.